### PR TITLE
Templates e2e testing

### PR DESCRIPTION
**Pull request by chrisFrazier77**

### CODE REVIEW

**Review by korgon — CHANGES_REQUESTED** [review]
Looking good!

File: packages/snap-preact/components/src/components/Atoms/TemplateSelect/TemplateSelect.tsx [hunk 1]
```diff
@@ -35,12 +35,19 @@ export const TemplateSelect = observer((properties: TemplateSelectProps): JSX.El
 		throw error;
 	}
 
+	let componentProp = {};
+	if (targeter.resultComponent && ResultComponent) {
+		componentProp = {
+			resultComponent: ResultComponent,
+		};
+	}
+
```
[the remark this answers is not in the document]
Comment (chrisFrazier77) [reply packages/snap-preact/components/src/components/Atoms/TemplateSelect/TemplateSelect.tsx:44]: yes

File: packages/snap-preact-demo/public/templates/index.html [hunk 2]
```diff
@@ -148,7 +148,7 @@
 
 
 				<div id="searchspring-layout">Content to be hidden!</div>
-
+				<div id="searchspring-content"></div>
```
[the remark this answers is not in the document]
Comment (chrisFrazier77) [reply packages/snap-preact-demo/public/templates/index.html:152]: its used for the branchOverrides test. I cant change the id of it unless we want to manually push up an updated branch bundle and change the selector in the target for that build.

File: packages/snap-preact-demo/tests/cypress/e2e/templates/searchHorizontal/query.cy.js
Comment (korgon) [inline packages/snap-preact-demo/tests/cypress/e2e/templates/searchHorizontal/query.cy.js:1]: This is a bit odd... moving from a template style config to a non-template style config...

File: packages/snap-preact-demo/tests/cypress/e2e/templates/search/query.cy.js
Comment (korgon) [inline packages/snap-preact-demo/tests/cypress/e2e/templates/search/query.cy.js:1]: This is a bit odd... moving from a template style config to a non-template style config...

**Review by korgon — COMMENTED** [review]
Looks good - though I think for the search and searchHorizontal tests we will want to find a better way to handle those unique configurations without resorting to using non-Snap Templates configs...

File: packages/snap-preact-demo/tests/cypress/e2e/templates/search/query.cy.js
Comment (korgon) [reply packages/snap-preact-demo/tests/cypress/e2e/templates/search/query.cy.js:1]: I'm an idiot.

File: packages/snap-preact-demo/tests/cypress/e2e/templates/searchHorizontal/query.cy.js
Comment (korgon) [reply packages/snap-preact-demo/tests/cypress/e2e/templates/searchHorizontal/query.cy.js:1]: I'm an idiot.

**Review by korgon — APPROVED** [review]
I'm an idiot.

**Pull request merged**